hw2/src/main/java/com/solvd/hw2/CustomPool.java

```diff
@@ -40,25 +55,49 @@ public static ArrayList<Connection> getIdleConns()
 
     public static synchronized Connection getConn() throws SQLException
     {
-        if (activeConns.size() == 0 && idleConns.size() == 0)
+        //Max size
+        if (activeConns.size() == maxConns)
+        {
+            //TODO: add multithread stuff
+            LOGGER.error("Max capacity reached. :(");
```

Comment:
One thing I do suggest in this `if statement` is:
- Instead of returning null if the max capacity has reached on a client trying to connect to the database, have the connection request wait until there's an available connection. That way, the request doesn't get ignored/discarded.
- To achieve this, you can use a `LinkedBlockingQueue<Connection>` instead of an ArrayList of connections, because a LinkedBlockingQueue supports that kind of logic.